src/java.desktop/share/classes/javax/swing/package-info.java

```diff
@@ -132,14 +132,14 @@
  * For overviews, tutorials, examples, guides, and other documentation,
  * please see:
  * <ul>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/articles-jsp-139072.html"
- *     target="_top">The Swing Connection</a></li>
+ *     <li><a href="https://www.oracle.com/java/technologies/a-swing-architecture.html"
+ *     target="_top">A Swing Architecture Overview</a></li>
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
- *     target="_top">Online Training</a>
+ *     <li><a href="https://www.oracle.com/java/technologies/javase/training-support.html"
+ *     target="_top">Java SE Training and Certification</a>
  *     at the Java Developer Connection <sup>SM</sup></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/index-jsp-142216.html"
+ *     <li><a href="https://www.oracle.com/java/technologies/java-foundation-classes.html"
  *     target="_top">Java Foundation Classes (JFC)</a> home page</li>
```

Comment:
Hm, I think this page is too outdated by now. It talks about JDK 1.4.0. It's completely irrelevant for JDK 19, is it?

Shall this link be dropped altogether?

Reply:
Maybe we can use https://www.oracle.com/java/technologies/foundation-classes-faq.html

Reply:
Yes, the page looks outdated.. I don't think any new content has been added post jdk1.4.2.
I can remove the link. 
In fact, I think the "JavaSE Training" link also is not relevant in this swing package, maybe it should be put in overall parent overview page.

Reply:
> In fact, I think the "JavaSE Training" link also is not relevant in this swing package, maybe it should be put in overall parent overview page.

Yeah, I agree it's not relevant for Swing.

Do you refer to [Java SE at a Glance](https://www.oracle.com/java/technologies/java-se-glance.html)?

Reply:
[Java SE Desktop Overview](https://www.oracle.com/java/technologies/javase/desktop-overview.html) is relevant as the overview of all desktop related technologies in Java. Yet it's not specifically about Swing. As such, this page fits for java.desktop module description better.

Reply:
> I guess in same way, it's more relevant to https://docs.oracle.com/en/java/javase/17/docs/api/index.html page

Yep!